src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -467,6 +469,30 @@ public IEnumerable<Attachment> GetAttachmentsForDocument(DocumentsOperationConte
             }
         }
 
+        public IEnumerable<Attachment> GetAttachmentsForDocument(DocumentsOperationContext context, AttachmentType type, LazyStringValue lowerDocumentId)
+        {
+            var table = context.Transaction.InnerTransaction.OpenTable(AttachmentsSchema, AttachmentsMetadataSlice);
+            using (From(context, lowerDocumentId, out Slice lowerDocumentIdSlice))
+            using (GetAttachmentPrefix(context, lowerDocumentIdSlice, type, Slices.Empty, out Slice prefixSlice))
+            {
+                foreach (var sr in table.SeekByPrimaryKeyPrefix(prefixSlice, Slices.Empty, 0))
+                {
+                    var attachment = TableValueToAttachment(context, ref sr.Value.Reader);
+                    if (attachment == null)
+                        continue;
+
+                    attachment.Size = GetAttachmentStreamLength(context, attachment.Base64Hash);
+
+                    yield return attachment;
+                }
+            }
+        }
+
+        private static ByteStringContext.InternalScope From(DocumentsOperationContext context, LazyStringValue lowerDocumentId, out Slice lowerDocumentIdSlice)
```

Comment:
Either rename this method or drop it and call `Slice.From` directly

Reply:
its unsafe method and you can't call unsafe from IEnumerable

Reply:
ok, makes sense